akka-cluster-sharding/src/main/scala/akka/cluster/sharding/ShardRegion.scala

```diff
@@ -179,7 +175,21 @@ object ShardRegion {
   /**
    * Reply to `GetCurrentRegions`
    */
-  @SerialVersionUID(1L) final case class CurrentRegions(regions: Set[Address]) {
+  @SerialVersionUID(2L) final case class CurrentRegions(regionActorPaths: Set[ActorPath]) {
```

Comment:
This is pretty much how I have thought about it and what is in the PR right now, it is possible to query the `ShardRegion` about what regions there are (internally forwarded request to the `ShardCoordinator`), which returns `ActorPath`s of all regions in the cluster, so that you (the user) can talk to each `ShardRegion` if you want.

Then there is the separate `GetCurrentShardRegionState`/`CurrentShardRegionState` which you can send to _one_ `ShardRegion` to get info about it's running entities. So, we do not provide the aggregate for the entire cluster ourselves - the user will have to do it - and as you say, hopefully think twice about it in their use case.